streams/src/main/java/org/apache/kafka/streams/kstream/internals/metrics/Sensors.java

```diff
@@ -39,4 +44,37 @@ public static Sensor lateRecordDropSensor(final InternalProcessorContext context
         );
         return sensor;
     }
+
+    public static Sensor recordLatenessSensor(final InternalProcessorContext context) {
+        final StreamsMetricsImpl metrics = context.metrics();
+
+        final Sensor sensor = metrics.nodeLevelSensor(
+            context.taskId().toString(),
+            context.currentNode().name(),
+            "record-lateness",
+            Sensor.RecordingLevel.INFO
+        );
+
+        final Map<String, String> tags = metrics.tagMap(
+            "task-id", context.taskId().toString(),
+            "processor-node-id", context.currentNode().name()
+        );
+        sensor.add(
+            new MetricName(
+                "record-lateness-avg",
+                "stream-processor-node-metrics",
+                "The average observed lateness of records.",
+                tags),
+            new Avg()
+        );
+        sensor.add(
+            new MetricName(
+                "record-lateness-max",
+                "stream-processor-node-metrics",
+                "The max observed lateness of records.",
+                tags),
+            new Max()
+        );
```

Comment:
The pre-canned Max-and-Avg function in StreamsMetricsImpl is specifically for latencies, which this is not. Hence adding the metrics right here.